generators/chipyard/src/main/scala/upf/UPFGraph.scala

```diff
@@ -0,0 +1,110 @@
+package chipyard.upf
+
+import freechips.rocketchip.diplomacy.LazyModule
+
+import scala.collection.mutable.{Map, Set, ListBuffer}
+
+case class PowerDomain (val name: String, val modules: ListBuffer[LazyModule], 
+                        val isTop: Boolean, val isGated: Boolean,
+                        val highVoltage: Double, val lowVoltage: Double) {
+    val mainVoltage = isGated match {
+        case true => highVoltage // gated nets should have access to high voltage rail (since they are being gated to optimize power)
+        case false => lowVoltage // currently assuming non-gated nets are on low voltage rail
+    }
+}
+
+class Node (var graph: PowerGraph, var nodeObj: Any) {
```

Comment:
Why can't we use a stdlib version of a graph?

Reply:
Changed implementation to use the standard scala graph, there wasn’t anything better with my own graph model.